### DISCUSSION TITLE
.zip file

### DISCUSSION THREAD

**Discussion by Krash-2504** [bazel-contrib/rules_python#2298] [Q&A]
Why does bazel create a .zip which is used to run the .exe and not a .exe which can run on its own? Is there any way to avoid this and create a standalone .exe?

**Comment by groodt** [comment]
It's a historical piece of functionality that has existed within bazel from before there were external rules.

It's also similar in nature to the [zipapp](https://docs.python.org/3/library/zipapp.html) functionality in the standard library.

Python is an interpreted language that also mixes with native platform dependent code. There isn't a standard format for shipping python executables. 

Container images are typically used as a convenient format for Web services and daemons. Zipapps can be useful for smaller tools that target a single platform. All that is needed is python on the host. 

You can look into more sophisticated solutions like PyOxidizer or Pex, but I'm less familiar with those approaches and they are more involved.
